### DISCUSSION TITLE
Estimating Suppressed QCEW Data

### DISCUSSION THREAD

**Discussion by matthewlchambers** [USEPA/flowsa#269] [Q&A] [answered]
Trying to adapt the `bls_clean_allocation_fba_w_sec()` function from `BLS_QCEW.py` to the new format, but I'm having a difficult time figuring out how this function is estimating the suppressed cells. Could someone briefly explain what's supposed to be happening in this function? Thanks!

**Comment by matthewlchambers** [comment, marked as answer]
I think this function accomplishes the desired goal. @catherinebirney or @bl-young , could you look it over and tell me what you think? Or if I've just misunderstood what we're trying to accomplish?

https://github.com/USEPA/flowsa/blob/f182edd12343fba3c7f9a9466852203a38b8a188/flowsa/data_source_scripts/temp_data_source_functions.py#L54-L122

**Reply by catherinebirney** [reply]
@matthewlchambers the bls_clean_allocation_fba_w_sec() calls on equally_allocate_suppressed_parent_to_child_naics(), which is a function used to estimate values for flow amounts that are 0 in the df - which is the way bls provides suppressed data. I can take a look at the code you provided later today or tomorrow. The function should be written generically so it can be applied to datasets other than bls

**Reply by matthewlchambers** [reply]
Once I feel sure my function is doing the right thing, then I'm happy to look at generalizing it. I just want to make sure that we're on the right track first. It was hard for me to follow `equally_allocate_suppressed_parent_to_child_naics()`, so I wrote the above function to do what's described in the function name. So, for each parent-children group (starting at the 2 digit level), it takes the parent total, subtracts the sum of non-suppressed children, then divides by the number of suppressed children, and attributes the result to each suppressed child. Is `equally_allocate_suppressed_parent_to_child_naics()` doing something beyond that?

**Reply by catherinebirney** [reply]
Your code looks right - `equally_allocate_suppressed_parent_to_child_naics()` isn't doing anything other than what you described.

**Reply by matthewlchambers** [reply]
Thanks, @catherinebirney! I don't think it should be too hard to generalize; I may have to come back to that task in a bit though.
